src/keydra/clients/github.py

```
@@ -0,0 +1,182 @@
+import requests
+
+from copy import deepcopy
+
+from base64 import b64encode
+
+from nacl import encoding, public
+
+
+API_URL = 'https://api.github.com'
+
+
+class GithubClient(object):
+    def __init__(self, user, passwd):
+        '''
+        Initializes a client for Github
+
+        :param user: Github username
+        :type user: :class:`str`
+        :param passwd: Github personal access token
+        :type passwd: :class:`passwd`
+        '''
+        self._authorizer = {
+            'Content-Type': 'application/json',
+            'Authorization': 'token {}'.format(passwd)
+        }
+
+    def _query(self, url, extraheaders=None):
+
+        if extraheaders:
+            headers = deepcopy(self._authorizer)
+            headers.update(extraheaders)
+        else:
+            headers = self._authorizer
+
+        resp = requests.get(url, headers=headers)
+
+        resp.raise_for_status()
+
+        try:
+            return resp.json()
+        except ValueError:
+            return resp.text
```

Hmm a generic function that can hit any endpointand return either json or string may result in some obscure errors.. ¯\\\_(ツ)_/¯

If we always expect valid json, I'd remove this try-catch and let the errors propagate. Otherwise I'd handle the conversion in the upstream callers that have enough context (i.e. the endpoint) to determine whether they return json or not.